src/inc/til.h

```diff
@@ -18,6 +18,216 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
+        {
+            return begin();
+        }
+
+        _NODISCARD _CONSTEXPR17 const_iterator cend() const noexcept
+        {
+            return end();
+        }
+
+        _NODISCARD _CONSTEXPR17 const_reverse_iterator crbegin() const noexcept
+        {
+            return rbegin();
+        }
+
+        _NODISCARD _CONSTEXPR17 const_reverse_iterator crend() const noexcept
+        {
+            return rend();
+        }
+
+        _NODISCARD constexpr size_type size() const noexcept
+        {
+            return _Used;
+        }
+
+        _NODISCARD constexpr size_type max_size() const noexcept
+        {
+            return _Size;
+        }
+
+        _NODISCARD constexpr bool empty() const noexcept
+        {
+            return !_Used;
+        }
+
+        _NODISCARD _CONSTEXPR17 reference at(size_type _Pos)
+        {
+            if (_Used <= _Pos)
+            {
+                _Xran();
+            }
+
+            return _Elems[_Pos];
+        }
+
+        _NODISCARD constexpr const_reference at(size_type _Pos) const
+        {
+            if (_Used <= _Pos)
+            {
+                _Xran();
+            }
+
+            return _Elems[_Pos];
+        }
+
+        _NODISCARD _CONSTEXPR17 reference operator[](_In_range_(0, _Used - 1) size_type _Pos) noexcept /* strengthened */
+        {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Pos < _Used, "some subscript out of range");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+
+            return _Elems[_Pos];
+        }
+
+        _NODISCARD constexpr const_reference operator[](_In_range_(0, _Used - 1) size_type _Pos) const
+            noexcept /* strengthened */
+        {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Pos < _Used, "some subscript out of range");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+
+            return _Elems[_Pos];
+        }
+
+        _NODISCARD _CONSTEXPR17 reference front() noexcept /* strengthened */
+        {
+            return _Elems[0];
+        }
+
+        _NODISCARD constexpr const_reference front() const noexcept /* strengthened */
+        {
+            return _Elems[0];
+        }
+
+        _NODISCARD _CONSTEXPR17 reference back() noexcept /* strengthened */
+        {
+            return _Elems[_Used - 1];
+        }
+
+        _NODISCARD constexpr const_reference back() const noexcept /* strengthened */
+        {
+            return _Elems[_Used - 1];
+        }
+
+        _NODISCARD _CONSTEXPR17 _Ty* data() noexcept
+        {
+            return _Elems;
+        }
+
+        _NODISCARD _CONSTEXPR17 const _Ty* data() const noexcept
+        {
+            return _Elems;
+        }
+
+        void push_back(const _Ty& _Val)
+        {
+            if (_Used >= _Size)
+            {
+                _Xran();
+            }
+
+            _Elems[_Used] = _Val;
+
+            ++_Used;
+        }
+
+        void pop_back() noexcept
+        {
+            if (_Used <= 0)

```

Comment:
One of the things that vector does is manages to invalidate its iterators when pop occurs. We're not doing that. Is it necessary before we merge this? (It really only works in debug mode anyway.)

Reply:
Not sure we need to do that. The iterators on vectors are stated by standard to be invalidated on pop; the check is only there as a CYA

Reply:
ok